jwst/dark_current/dark_sub.py

```diff
@@ -178,23 +275,38 @@ def subtract_dark(input, dark):
 
     """
 
+    dark_ints = dark.data.shape[0] -1
+
     log.debug("subtract_dark: nints=%d, ngroups=%d, size=%d,%d",
               input.meta.exposure.nints, input.meta.exposure.ngroups,
               input.data.shape[-1], input.data.shape[-2])
 
+
     # Create output as a copy of the input science data model
     output = input.copy()
 
+    
     # combine the science and dark DQ arrays
-    output.pixeldq = np.bitwise_or(input.pixeldq, dark.dq)
+    detector = input.meta.instrument.detector
+    if(detector[:3] == 'MIR'):
+           output.pixeldq = np.bitwise_or(input.pixeldq, np.bitwise_or(dark.dq[0,0,:,:], dark.dq[1,0,:,:]))
```

Comment:
In order to avoid the explicit assumption that dark.dq has a dimension of 2 along the first axis, you might be able to do something like:

 output.pixeldq = np.bitwise_or(input.pixeldq, np.bitwise_or(dark.dq[:,0], axis=0))

where the np.bitwise_or(dark.dq, axis=0) first just does an OR along axis 0, the integration axis, however long that axis happens to be, resulting in a 2D array of "collapsed" flag values, which then gets OR'd with the 2D input.pixeldq array.